app/lib/search/mem_index.dart

```diff
@@ -225,11 +225,27 @@ class InMemoryPackageIndex {
     final textResults = _searchText(
       packageScores,
       parsedQueryText,
-      includeNameMatches: (query.offset ?? 0) == 0,
       textMatchExtent: query.textMatchExtent ?? TextMatchExtent.api,
     );
 
-    final nameMatches = textResults?.nameMatches;
+    String? bestNameMatch;
+    if (parsedQueryText != null) {
+      final matches = _packageNameIndex.lookupMatchingNames(parsedQueryText);
```

Comment:
Not sure if this is already how it works, but can we make it such that if there are multiple matches, and one is an exact match, then we use it regardless of download-count?

Reply:
Fair point, fixed.
Note: I think the multiple matches are mostly theoretical, e.g. `a_bc` and `ab_c` would be matched for the query `abc`, however, `pub.dev` wouldn't allow such package names in the first place (unless they were exempted), and then we would fall back to the download counts.